src/site/content/en/blog/eventing-deepdive/index.md

```diff
@@ -0,0 +1,541 @@
+    e.stopImmediatePropagation();
+  },
+  false,
+);
+
+document.getElementById('A').addEventListener(
+  'click',
+  function (e) {
+    console.log('When #A is clicked, I would have run third, if not for stopImmediatePropagation');
+  },
+  false,
+);
+```
+
+The above example will result in the following console output:
+
+```bash
+"When #A is clicked, I shall run first!"
+"When #A is clicked, I shall run second!"
+```
+
+Note that the third event handler never runs due to the fact that the second event handler calls
+`e.stopImmediatePropagation()`.
+
+### `event.preventDefault()`
+
+If `stopPropagation()` prevents an event from traveling "downwards" (capturing) or "upwards"
+(bubbling), what then, does `preventDefault()` do? It sounds like it does something similar. Does
+it?
+
+Not really. While the two are often confused, they actually don't have much to do with each other.
+When you see `preventDefault()`, in your head, add the word "action." Think "prevent the default
+action."
+
+And what is the default action you may ask? Unfortunately, the answer to that isn't quite as clear
+because it's highly dependent on the element + event combination in question. And to make matters
+even more confusing, sometimes there is no default action at all!
+
+Let's begin with a very simple example to understand. What do you expect to happen when you click a
+link on a web page? Obviously, you expect the browser to navigate to the URL specified by that link.
+In this case, the element is an anchor tag and the event is a click event. That combination (`<a>` +
+`click`) has a "default action" of navigating to the link's href. What if you wanted to _prevent_
+the browser from performing that default action? That is, suppose you want to prevent the browser
+from navigating to the URL specified by the `<a>` element's `href` attribute? This is what
+`preventDefault()` will do for you. Consider this example:
+
+```html
+<a id="avett" href="http://www.theavettbrothers.com">The Avett Brothers</a>
+```
+
+```js
+document.getElementById('avett').addEventListener(
+  'click',
+  function (e) {
+    e.preventDefault();
+    console.log('Maybe we should just play some of their music right here instead?');
+  },
+  false,
+);
+```
+
+Normally, clicking the link labelled The Avett Brothers would result in browsing to
+`www.theavettbrothers.com`. In this case though, we've wired up a click event handler to the `<a>`
+element and specified that the default action should be prevented. Thus, when a user clicks this
+link, they won't be navigated anywhere, and instead the console will simply log "Maybe we should
+just play some of their music right here instead?"
+
+What other element/event combinations allow you to prevent the default action? I cannot possibly
+list them all, and sometimes you have to just experiment to see. But briefly, here are a few:
+
+- `<form> element + submit event`: `preventDefault()` for this combination will prevent a form from
```

Comment:
Should the left side of these points be in backticks? I can see the element being valid, but the event itself is a "string", so perhaps:

```md
- `<form>` element + "submit" event: `preventDefault()` for this ...
```

Reply:
Good idea. I have updated to put only the html or js in backticks. I've also quoted the string event names. Hopefully it's clearer now.